From 0f2e3ecda67099444039092c14ff9630235bd631 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Mon, 21 Dec 2015 19:39:52 +0530
Subject: [PATCH 02/14] USBC-PE: Fix reset issue on source cap tx fail.

As per PD spec v1.1, source port should not send hardreset if
no response (gcrc) from port partner. Instead src port has to
retry the src_cap for nCapsCount times.

Change-Id: I1e8d88214334a2206cbe6ea913714336d1618d61
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15496
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/451045
---
 drivers/usb/typec/pd/policy_engine.c | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/drivers/usb/typec/pd/policy_engine.c b/drivers/usb/typec/pd/policy_engine.c
index 0af7cd4..b8c0006 100644
--- a/drivers/usb/typec/pd/policy_engine.c
+++ b/drivers/usb/typec/pd/policy_engine.c
@@ -279,6 +279,9 @@ static void pe_handle_gcrc_received(struct policy_engine *pe)
 			pe_cancel_timer(pe, NO_RESPONSE_TIMER);
 		pe->hard_reset_counter = 0;
 		pe->src_caps_couner = 0;
+		/* Start sender response timer */
+		pe_start_timer(pe, SENDER_RESPONSE_TIMER,
+					PE_TIME_SENDER_RESPONSE);
 		break;
 
 	case PE_SRC_NEGOTIATE_CAPABILITY:
@@ -1381,11 +1384,8 @@ static void pe_timer_expire_worker(struct work_struct *work)
 
 	case CRC_RECEIVE_TIMER:
 		if (pe->cur_state == PE_SRC_SEND_CAPABILITIES) {
-			pe_cancel_timer(pe, SENDER_RESPONSE_TIMER);
-			if (!pe->is_pp_pd_capable) {
-				pe_change_state(pe, PE_SRC_DISCOVERY);
-				break;
-			}
+			pe_change_state(pe, PE_SRC_DISCOVERY);
+			break;
 		} else if (pe->cur_state == PE_PRS_SRC_SNK_WAIT_SOURCE_ON) {
 			log_err("PS_RDY Sent fail during pr_swap");
 			pe_change_state(pe, PE_ERROR_RECOVERY);
@@ -1796,9 +1796,6 @@ pe_process_state_pe_src_send_capabilities(struct policy_engine *pe)
 		pe_change_state(pe, PE_SRC_DISCOVERY);
 		return;
 	}
-	/* SrcCap sent successfuly, start sender response timer*/
-	pe_start_timer(pe, SENDER_RESPONSE_TIMER, PE_TIME_SENDER_RESPONSE);
-
 }
 
 static void
-- 
1.9.1

